pkg/agent/ippool/event.go

```diff
@@ -117,8 +114,11 @@ func (e *EventHandler) EventListener() (err error) {
 
 	controller := NewController(queue, indexer, informer, e.poolRef, e.dhcpAllocator, e.poolCache)
 	stop := make(chan struct{})
-	defer close(stop)
+
 	go controller.Run(1, stop)
 
-	select {}
+	<-ctx.Done()
+	controller.Stop(stop)
+
+	logrus.Info("(eventhandler.Run) IPPool event listener terminated")
```

Comment:
Because the `stop` is only used in the controller, so you probably could wrap it into controller code when you new the contoller.

Reply:
I moved the channel into the `Controller` struct. PTAL, thanks!